### PR TITLE
CGMES condensers shall not have generating units

### PR DESCRIPTION
**Pull request by zamarrenolm**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

In CGMES, all synchronous machines are expected to have an associated generating unit.

**What is the new behavior (if this is a feature change)?**
Synchronous condensers have no capability for active power output. They shall not be associated with a generating unit.
It is stated in ENTSO-E QoCDC rules. Rule: SynchronousCondenser Level: 3 Severity: ERROR.
Note: this is also stated in the CGMES 2.4.15 profile (IEC-61970-600-2), section 3.9.47 SynchronousMachine: If a synchronous condenser is being modeled so that there is no capability for real power output, the SynchronousMachine is not required to be associated with a GeneratingUnit. In this case, the type and operatingMode attributes must both be set to “condenser”.

**Does this PR introduce a breaking change or deprecate an API?**
- [X] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [X] The *Breaking Change* or *Deprecated* label has been added
- [X] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

We have deprecated the method to obtain the synchronous machines (SMs) in the `CgmesModel` class: `synchronousMachines()` , and  defined two new methods to be used from now on: `synchronousMachinesGenerators()` to get SMs that can act as generators and `synchronousMachinesCondensers()` that will return only the SMs that are of type `condenser`.

**Other information**:

The Svedala test configuration of ENTSO-E CAS v3.0.2 contains a condenser that has an associated generating unit with `minP = maxP = 0`. The QoCDC rule is not respected in this test case. The PowSyBl CGMES converter will import the condenser, ignoring the association with the generating unit.

### CODE REVIEW

**Comment by annetill** [comment]
Note: export without GU to be compliant with QoCDC

**Review by annetill — APPROVED** [review]
Export must be compliant with QoCDC: no generating unit for condensers.

**Comment by zamarrenolm** [comment]
> Export must be compliant with QoCDC: no generating unit for condensers.

adapted and tested the export

File: cgmes/cgmes-model/src/main/resources/CIM100.sparql [hunk 1]
```diff
@@ -76,7 +80,7 @@ OPTIONAL { GRAPH ?graphSSH {
         cim:SynchronousMachine.referencePriority ?referencePriority ;
         cim:RegulatingCondEq.controlEnabled ?controlEnabled .
 }}
-OPTIONAL { GRAPH ?graphSSH {
+OPTIONAL { GRAPH ?graphSSH2 {
```
[the remark this answers is not in the document]
Comment (zamarrenolm) [reply cgmes/cgmes-model/src/main/resources/CIM100.sparql:83]: To have the same semantics that we have in `CIM16.sparql`. Here we are just saying that SSH attributes for the synchronous machine and generating unit are not necessarily read from the same SSH instance file.

**Review by rcourtier — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/test/resources/issues/condenser2_EQ.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    <cim:Terminal rdf:ID="_Condenser1_T">
+        <cim:IdentifiedObject.name>Terminal</cim:IdentifiedObject.name>
+        <cim:ACDCTerminal.sequenceNumber>1</cim:ACDCTerminal.sequenceNumber>
+        <cim:Terminal.phases rdf:resource="http://iec.ch/TC57/CIM100#PhaseCode.ABC" />
+        <cim:Terminal.ConductingEquipment rdf:resource="#_Condenser1" />
+        <cim:Terminal.ConnectivityNode rdf:resource="#_ConnectivityNode1" />
+    </cim:Terminal>
+    <cim:RegulatingControl rdf:ID="_Condenser1_RC">
+        <cim:IdentifiedObject.name>Condenser 1 Regulating Control</cim:IdentifiedObject.name>
+        <cim:RegulatingControl.mode rdf:resource="http://iec.ch/TC57/CIM100#RegulatingControlModeKind.voltage" />
```
[the remark this answers is not in the document]
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/test/resources/issues/condenser2_EQ.xml]: applied the correction

File: cgmes/cgmes-conversion/src/test/resources/issues/condenser2_EQ.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        <cim:IdentifiedObject.name>Condenser 1</cim:IdentifiedObject.name>
+        <cim:Equipment.EquipmentContainer rdf:resource="#_VoltageLevel1" />
+        <cim:RegulatingCondEq.RegulatingControl rdf:resource="#_Condenser1_RC" />
+        <cim:RotatingMachine.ratedPowerFactor>0.95</cim:RotatingMachine.ratedPowerFactor>
+        <cim:RotatingMachine.ratedS>300</cim:RotatingMachine.ratedS>
+        <cim:RotatingMachine.ratedU>400</cim:RotatingMachine.ratedU>
+        <cim:SynchronousMachine.maxQ>0</cim:SynchronousMachine.maxQ>
+        <cim:SynchronousMachine.minQ>-110</cim:SynchronousMachine.minQ>
+        <cim:SynchronousMachine.qPercent>100</cim:SynchronousMachine.qPercent>
+        <cim:SynchronousMachine.type rdf:resource="http://iec.ch/TC57/CIM100#SynchronousMachineKind.condenser" />
```
[the remark this answers is not in the document]
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/test/resources/issues/condenser2_EQ.xml]: applied the correction

**Review by rcourtier — APPROVED** [review]

**Pull request merged**